rtpghi~-help.pd

```diff
@@ -0,0 +1,257 @@
+#X connect 9 1 13 0;
+#X connect 10 0 12 1;
+#X connect 11 0 12 0;
+#X connect 12 0 5 0;
+#X connect 13 0 14 0;
+#X connect 14 0 11 1;
+#X connect 14 1 10 1;
+#X restore 346 191 pd fft-irfft;
+#X obj 347 263 dac~;
+#N canvas 618 124 760 628 fft-rtpghi 1;
+#X obj 37 232 rfft~;
+#X obj 36 255 *~;
+#X obj 70 254 *~;
+#X obj 35 315 sqrt~;
+#X obj 145 77 block~ 1024 4;
+#X obj 81 479 rifft~;
+#X obj 145 99 table \$0.hann;
+#X obj 409 195 s \$0.hann;
+#X msg 408 170 cosinesum \$1 0.5 -0.5;
+#X obj 62 139 tabreceive~ \$0.hann;
+#X obj 33 140 *~;
+#X obj 145 53 t a a;
+#X obj 252 77 unpack s f f f;
+#X obj 130 567 tabreceive~ \$0.hann;
+#X obj 101 566 *~;
+#X obj 38 41 inlet~;
+#X obj 105 602 outlet~;
+#X obj 44 189 *~ 1;
+#X obj 147 28 inlet;
+#X obj 35 281 +~;
+#X obj 252 122 t f f;
+#N canvas 915 136 564 537 normalize 0;
+#X obj 119 39 inlet;
+#X obj 118 442 outlet;
+#X obj 300 44 inlet;
+#X obj 193 282 /;
+#X obj 216 231 t b f;
+#X obj 207 130 sel 32 16 8 4 2;
+#X obj 243 159 1;
+#X obj 270 158 2;
+#X obj 299 158 4;
+#X obj 330 158 8;
+#X obj 196 161 0.816;
+#X floatatom 174 194 5 0 0 0 - - - 0;
+#X obj 139 234 * 1.55;
+#X connect 0 0 12 0;
+#X connect 2 0 5 0;
+#X connect 3 0 1 0;
+#X connect 4 0 3 0;
+#X connect 4 1 3 1;
+#X connect 5 0 10 0;
+#X connect 5 1 6 0;
+#X connect 5 2 7 0;
+#X connect 5 3 8 0;
+#X connect 5 4 9 0;
+#X connect 6 0 4 0;
+#X connect 7 0 4 0;
+#X connect 8 0 4 0;
+#X connect 9 0 4 0;
+#X connect 10 0 4 0;
+#X connect 11 0 12 1;
+#X connect 12 0 3 0;
+#X restore 248 212 pd normalize;
+#X obj 248 235 t f f;
+#X obj 77 397 /~;
+#X obj 117 400 /~;
+#X obj 35 344 rtpghi~ 16 1e-05 0 hann;
+#X msg 123 263 overlap 8;
+#X connect 0 0 1 0;
+#X connect 0 0 1 1;
+#X connect 0 1 2 0;
+#X connect 0 1 2 1;
+#X connect 1 0 19 0;
+#X connect 2 0 19 1;
+#X connect 3 0 25 0;
+#X connect 5 0 14 0;
+#X connect 8 0 7 0;
+#X connect 9 0 10 1;
+#X connect 10 0 17 0;
+#X connect 11 0 4 0;
+#X connect 11 1 12 0;
+#X connect 12 1 20 0;
+#X connect 12 2 21 1;
+#X connect 13 0 14 1;
+#X connect 14 0 16 0;
+#X connect 15 0 10 0;
+#X connect 17 0 0 0;
+#X connect 18 0 11 0;
+#X connect 19 0 3 0;
+#X connect 20 0 21 0;
+#X connect 20 1 8 0;
+#X connect 21 0 22 0;
+#X connect 22 0 23 1;
+#X connect 22 1 24 1;
+#X connect 23 0 5 0;
+#X connect 24 0 5 1;
+#X connect 25 0 23 0;
+#X connect 25 1 24 0;
+#X connect 26 0 25 0;
+#X restore 43 187 pd fft-rtpghi;
+#X obj 67 123 tgl 15 0 empty empty empty 17 7 0 10 #fcfcfc #000000
+#000000 0 1;
+#X obj 48 146 *~ 1;
+#X obj 393 154 tgl 15 0 empty empty empty 17 7 0 10 #fcfcfc #000000
+#000000 0 1;
+#X obj 343 152 *~ 1;
+#X obj 86 231 env~;
+#X floatatom 86 254 5 0 0 0 - - - 0;
+#X floatatom 385 264 5 0 0 0 - - - 0;
+#X obj 385 240 env~;
+#X obj 671 263 dac~;
+#X obj 706 207 tgl 15 0 empty empty empty 17 7 0 10 #fcfcfc #000000
+#000000 0 1;
+#X obj 666 209 *~ 1;
+#X floatatom 711 263 5 0 0 0 - - - 0;
+#X obj 711 239 env~;
+#X obj 47 96 noise~;
+#X obj 340 95 noise~;
+#X obj 664 131 noise~;
+#X msg 366 30 set 2048 16 1;
```

Comment:
thanks for the much improved help-patch.

why do you explicitly set the blocskize/overlap/upsampling with a loadbanged message, when you could just use `[block~ 2048 16]` in the relevant subpatches?

it would be cool to have a:
- one radiobutton to select the reconstruction algorithm (phaseret, fft, none)
- one radiobutton to select the input signal
  (rather than multiple toggles that allow you to listen to the phaseret/noise at the same time as sine/fft and sine/none)
- have a musical input signal (e.g. by allowing the user to select a soundfile) - sines and noise typically do not allow you to assess the quality of an FFT-algorithm
- have a master fader, so you don't blow the PA